zppy/templates/coupled_global.py

```diff
@@ -132,6 +132,33 @@ def get_ylim(standard_range, extreme_values):
 # -----------------------------------------------------------------------------
 # Plotting functions
 
+# Generic plot function
+def plot_generic(ax, xlim, exps, var_name):
+    param_dict = {
+        "2nd_var": False,
+        "axhline_y": 0,
+        "check_exp_ocean": False,
+        "check_exp_vol": False,
+        "check_exp_year": True,
+        "default_ylim": [-1, 1],
+        "do_add_line": True,
+        "do_add_trend": True,
+        "format": "%4.2f",
+        "lw": 1.0,
+        "ohc": False,
+        "set_axhline": False,
+        "set_legend": True,
+        "shorten_year": False,
+        "title": var_name,
+        "use_getmoc": False,
+        "var": lambda exp: np.array(exp["annual"][var_name]),
+        "verbose": False,
+        "vol": False,
+        "ylabel": "???",
```

Comment:
No way to know the units unless we were to have users pass in a tuple of `(var_name, ylabel)`

Reply:
Perhaps it would be possible to extract the units once we get the variable from the `v = ts.globalAnnual(var)` line?

Reply:
yes, the var_name and units should preserve in cdms2's TransientVariable (in current implementation)  or xarry's DataArray (after migrating to xarray). I suspect that the units were stripped of in readTS.py, and it should possible to get it back, i think.

Reply:
Yes, my latest commit https://github.com/E3SM-Project/zppy/pull/400/commits/5fd205533c472cf91131c7ffc74a1aa88885d876 appears to allow units to be deduced automatically.